### ISSUE TITLE
Add autocorrect for undeclared instance variables

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#3514]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strict%0A%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20X%20%3D%20Object.new%0A%0A%20%20sig%20%7Bvoid%7D%0A%20%20def%20initialize%0A%20%20%20%20%40y%20%3D%20Object.new%0A%20%20end%0Aend)

```ruby
# typed: strict

class A
  extend T::Sig

  X = Object.new

  sig {void}
  def initialize
    @y = Object.new
  end
end
```

#### Observed output

```
editor.rb:10: Use of undeclared variable @y https://srb.help/6002
    10 |    @y = Object.new
            ^^

editor.rb:6: Constants must have type annotations with T.let when specifying # typed: strict https://srb.help/7027
     6 |  X = Object.new
              ^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    editor.rb:6: Replace with T.let(Object.new, Object)
     6 |  X = Object.new
              ^^^^^^^^^^
Errors: 2
```

#### Expected behavior

We should have an autocorrect (and thus a quickfix) to add a `T.let` annotation for the `@y = Object.new` the same way that we have one for `X = Object.new` (constants).

- - -

- This should only suggest `T.let` inside `initialize` methods or the top-level of a class body.
- It's probably too hard to write an autocorrect for a nilable variable inside a non-initialize method, but if whoever takes this wants to try to do that too, go for it.

**Comment by jez** [comment]
I implemented this recently

**Comment by jez** [comment]
Fixed in #6016
